openmessaging/src/test/java/io/openmessaging/rocketmq/consumer/LocalMessageCacheTest.java

```diff
@@ -86,4 +95,77 @@ public void testSubmitConsumeRequest() throws Exception {
         localMessageCache.submitConsumeRequest(consumeRequest);
         assertThat(localMessageCache.poll()).isEqualTo(consumedMsg);
     }
+
+    @Test
+    public void testBatchPollMessage() throws Exception {
+        byte[] body = new byte[] {'1', '2', '3'};
+        MessageExt consumedMsg = new MessageExt();
+        consumedMsg.setMsgId("NewMsgId");
+        consumedMsg.setBody(body);
+        consumedMsg.putUserProperty(NonStandardKeys.MESSAGE_DESTINATION, "TOPIC");
+        consumedMsg.setTopic("HELLO_QUEUE");
+
+        byte[] body1 = new byte[] {'4', '5', '6'};
+        MessageExt consumedMsg1 = new MessageExt();
+        consumedMsg1.setMsgId("NewMsgId1");
+        consumedMsg1.setBody(body1);
+        consumedMsg1.putUserProperty(NonStandardKeys.MESSAGE_DESTINATION, "TOPIC");
+        consumedMsg1.setTopic("HELLO_QUEUE1");
+
+        when(consumeRequest.getMessageExt()).thenReturn(consumedMsg);
+        when(consumeRequest1.getMessageExt()).thenReturn(consumedMsg1);
+        localMessageCache.submitConsumeRequest(consumeRequest);
+        localMessageCache.submitConsumeRequest(consumeRequest1);
+        KeyValue properties = new DefaultKeyValue();
+        properties.put(NonStandardKeys.TIMEOUT, 3000);
+        List<MessageExt> messageExts = localMessageCache.batchPoll(properties);
+        assertThat(messageExts.size()).isEqualTo(2);
+        MessageExt messageExt1 = null;
+        MessageExt messageExt2 = null;
+        for (MessageExt messageExt : messageExts) {
+            if (messageExt.getMsgId().equals("NewMsgId")) {
+                messageExt1 = messageExt;
+            }
+            if (messageExt.getMsgId().equals("NewMsgId1")) {
+                messageExt2 = messageExt;
+            }
+        }
+        assertThat(messageExt1).isNotNull();
```

Comment:
Here,I think it will be better if you can verify the messageExt's detailed content values by using assertEquals method.For example,you can verify body value,UserProperty and topic,other places you can refer to modify if the same as here. @odbozhou

Reply:
This has increased verification